app/assets/javascripts/controllers/schedule/schedule_form_controller.js

```diff
@@ -234,6 +251,9 @@ ManageIQ.angular.app.controller('scheduleFormController', ['$http', '$scope', 's
       $http.post('/ops/automate_schedules_set_vars/' + scheduleFormId)
         .then(postAutomateSchedulesSetVarsComplete)
         .catch(miqService.handleFailure);
+
+      $scope.buildZonesList();
+      checkFormDataRetrieval($scope);
```

Comment:
this won't work right, because `postAutomateSchedulesSetVarsComplete` is already doing that, and there's no ordering guarantee between the 2 requests.

You would need to add the promise returned by the `$http.post` above to the all promises array and wait for that.